inference-engine/thirdparty/clDNN/src/include/loop_inst.h

```diff
@@ -173,54 +228,73 @@ struct typed_program_node<loop> : public typed_program_node_base<loop> {
         }
     }
 
-    void process_single_int_input(const primitive_id& id) const {
+    void process_current_iteration() const {
+        const primitive_id& current_iteration_id = get_current_iteration_id();
+        if (current_iteration_id.empty()) {
+            return;
+        }
+
         const topology_map& body_topology_map = body.get_primitives();
-        if (!id.empty()) {
-            // add input_layout if not exist
-            if (body_topology_map.count(id)) {
-                layout body_input_layout(data_types::i32, format::bfyx, {1, 1, 1, 1});
-                body.add(std::make_shared<input_layout>(id, body_input_layout));
+        const layout body_input_layout(data_types::i64, format::bfyx, {1, 1, 1, 1});
+
+        // add current_iteration primitive
+        // if current_iteration primitive is not exist in body,
+        if (body_topology_map.find(current_iteration_id) == body_topology_map.end()) {
+            body.add(std::make_shared<input_layout>(current_iteration_id, body_input_layout));
+        } else {
+            const auto& body_input_prim = body.at(current_iteration_id);
+            const auto input_layout_prim = std::dynamic_pointer_cast<input_layout>(body_input_prim);
+            if (!input_layout_prim) {
+                CLDNN_ERROR_MESSAGE(this->id(), "Not cldnn::input_layout");
             } else {
-                const auto& body_input_prim = body.at(id);
-                CLDNN_ERROR_BOOL(this->id(), "Error while building body program",
-                    body_input_prim->type != input_layout::type_id(),
-                    id + " is not cldnn::input_layout");
-                const auto input_layout_prim = static_cast<const input_layout*>(body_input_prim.get());
-                CLDNN_ERROR_BOOL(this->id(), "Error while building body program",
-                    !static_cast<bool>(input_layout_prim->output_data_type),
-                    "data_type of " + id + " is not specified");
-                CLDNN_ERROR_BOOL(this->id(), "Error while building body program",
-                    !is_integer(*input_layout_prim->output_data_type),
-                    id + " is not integer type");
-                CLDNN_ERROR_BOOL(this->id(), "Error while building body program",
-                    input_layout_prim->layout.count() != 1,
-                    id + " should have 1 element");
+                input_layout_prim->change_layout(body_input_layout);
             }
         }
+
+        // add increment value: 1
```

Comment:
...seems this should be put above

Reply:
I think that this comment is trying to describe the data used to increment current_iteration by 1.
The location of this comment is correct but it seems to be an expression that can bring confusion.
I will fix it. thanks.